net/peer.go

```diff
@@ -629,10 +627,16 @@ func (p *Peer) handleDocCreateLog(evt events.Update) error {
 		return errors.Wrap("failed to get DocKey from broadcast message", err)
 	}
 
+	// We need to register the document before pushing to the replicators if we want to
+	// ensure that we have subscribed to the topic.
+	err = p.RegisterNewDocument(p.ctx, dockey, evt.Cid, evt.Block, evt.SchemaID)
```

Comment:
question: This change is not noted in the description, is it still needed, or is this left here accidentally?

Reply:
I think it's still needed for the situation where the target would receive the push to replicator and then the target getting an update before the source registers to the topic. This probably would never happen in real life but it did happen in the tests.